services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -658,17 +659,28 @@ protected void initDasCustody() {
     if (!spec.isMilestoneSupported(SpecMilestone.EIP7594)) {
       return;
     }
-    DataColumnSidecarDBImpl sidecarDB =
-        new DataColumnSidecarDBImpl(
-            combinedChainDataClient, eventChannels.getPublisher(SidecarUpdateChannel.class));
+    SpecConfigEip7594 configEip7594 =
+        SpecConfigEip7594.required(spec.forMilestone(SpecMilestone.EIP7594).getConfig());
+    final DataColumnSidecarDbAccessor dbAccessor;
+    {
```

Comment:
Why do you init it this way?

Reply:
Do you mean the separate block? It's just to keep the `db` instance in local scope and doesn't leak it to other components accidentally. All the db access should go through the accessor